src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleEqualityTests.cs

```diff
@@ -534,43 +534,22 @@ static bool M((int?, bool?) t1)
             comp.VerifyDiagnostics();
 
             comp.VerifyIL("C.M", @"{
-  // Code size       63 (0x3f)
+  // Code size       32 (0x20)
   .maxstack  2
-  .locals init (System.ValueTuple<int?, bool?> V_0,
-                int? V_1,
-                int V_2,
-                bool? V_3,
-                bool V_4)
+  .locals init (System.ValueTuple<int?, bool?> V_0)
   IL_0000:  ldarg.0
   IL_0001:  stloc.0
-  IL_0002:  ldloc.0
-  IL_0003:  ldfld      ""int? System.ValueTuple<int?, bool?>.Item1""
-  IL_0008:  stloc.1
-  IL_0009:  ldc.i4.2
-  IL_000a:  stloc.2
-  IL_000b:  ldloca.s   V_1
-  IL_000d:  call       ""int int?.GetValueOrDefault()""
-  IL_0012:  ldloc.2
-  IL_0013:  ceq
-  IL_0015:  ldloca.s   V_1
-  IL_0017:  call       ""bool int?.HasValue.get""
-  IL_001c:  and
-  IL_001d:  brfalse.s  IL_003d
-  IL_001f:  ldloc.0
-  IL_0020:  ldfld      ""bool? System.ValueTuple<int?, bool?>.Item2""
-  IL_0025:  stloc.3
-  IL_0026:  ldc.i4.1
-  IL_0027:  stloc.s    V_4
-  IL_0029:  ldloca.s   V_3
-  IL_002b:  call       ""bool bool?.GetValueOrDefault()""
-  IL_0030:  ldloc.s    V_4
-  IL_0032:  ceq
-  IL_0034:  ldloca.s   V_3
-  IL_0036:  call       ""bool bool?.HasValue.get""
-  IL_003b:  and
-  IL_003c:  ret
-  IL_003d:  ldc.i4.0
-  IL_003e:  ret
+  IL_0002:  ldloca.s   V_0
+  IL_0004:  ldflda     ""int? System.ValueTuple<int?, bool?>.Item1""
+  IL_0009:  call       ""int int?.GetValueOrDefault()""
+  IL_000e:  ldc.i4.2
+  IL_000f:  bne.un.s   IL_001e
+  IL_0011:  ldloca.s   V_0
+  IL_0013:  ldflda     ""bool? System.ValueTuple<int?, bool?>.Item2""
+  IL_0018:  call       ""bool bool?.GetValueOrDefault()""
```

Comment:
Is this a breaking change for a scenario when ```bool?``` value is 2, for example?

Reply:
I believe not. See https://github.com/dotnet/roslyn/issues/66431#issuecomment-1822751507.

Reply:
> I believe not.

I am not so sure about that.

Reply:
Edit: yes, you're right, it's a breaking change because the result will be different, but only for `bool`s represented differently than 0 or 1 (as discussed in the issue).

Reply:
Note: I would not describe this as a breaking change. It is simply depending on undefined behavior.